sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/GroupStateImpl.scala

```diff
@@ -178,7 +179,7 @@ private[sql] class GroupStateImpl[S] private(
       throw new IllegalArgumentException(s"Provided duration ($duration) is not positive")
     }
 
-    val millisPerMonth = CalendarInterval.MICROS_PER_DAY / 1000 * 31
+    val millisPerMonth = TimeUnit.MICROSECONDS.toMillis(CalendarInterval.MICROS_PER_DAY) * 31
```

Comment:
Can this just be `MONTHS.toMillis(1)`? or just make the conversion in the next line directly?

Reply:
Unfortunately, there is no `MONTHS.toMillis`. 

> or just make the conversion in the next line directly?

It seems there is no way to perform the conversion directly because conversion of milliseconds to "estimated" chrono units (starting from days) is not supported in `java.time`

Reply:
Oh right of course. Hm, let's not change the logic here, but are the semantics of a 'month' interval even meaningful! That's why CalendarInterval breaks it out.